src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -11,65 +11,50 @@
 
 
 public class DriveSubsystem extends SubsystemBase {
-    private CANSparkMax mLeft1;
-    private CANSparkMax mLeft2;
-    private CANSparkMax mRight1;
-    private CANSparkMax mRight2;
+    private static CANSparkMax mLeft1 = new CANSparkMax(Constants.DriveLeft1, MotorType.kBrushless);
+    private static CANSparkMax mLeft2 = new CANSparkMax(Constants.DriveLeft2, MotorType.kBrushless);
+    private static CANSparkMax mLeft3 = new CANSparkMax(Constants.DriveLeft3, MotorType.kBrushless);
+    private static CANSparkMax mRight1 = new CANSparkMax(Constants.DriveRight1, MotorType.kBrushless);
+    private static CANSparkMax mRight2 = new CANSparkMax(Constants.DriveRight2, MotorType.kBrushless);
+    private static CANSparkMax mRight3 = new CANSparkMax(Constants.DriveRight3, MotorType.kBrushless);
     private CANPIDController l_pidController;
     private CANPIDController r_pidController;
     private CANEncoder l_encoder;
     private CANEncoder r_encoder;
     private DifferentialDrive differentialDrive;
-    public double kP, kI, kD, kIz, kFF, kMaxOutput, kMinOutput, maxRPM;
+    public double kP = Constants.kPDrive, kI = Constants.kIDrive, kD = Constants.kDDrive, kIz = Constants.kIzDrive, kFF = Constants.kFFDrive,
+     kMaxOutput = Constants.kMaxOutputDrive, kMinOutput = Constants.kMinOutputDrive;
     private static final Logger LOGGER = Logger.getLogger(DriveSubsystem.class.getName());
 
   public DriveSubsystem() {
-    mLeft1 = new CANSparkMax(1, MotorType.kBrushless);
-    mLeft2 = new CANSparkMax(2, MotorType.kBrushless);
-    mRight1 = new CANSparkMax(3, MotorType.kBrushless);
-    mRight2 = new CANSparkMax(4, MotorType.kBrushless);
-    mLeft1.restoreFactoryDefaults();
-    mLeft2.restoreFactoryDefaults();
-    mRight1.restoreFactoryDefaults();
-    mRight2.restoreFactoryDefaults();
-    mLeft1.enableVoltageCompensation(12);
-    mLeft2.enableVoltageCompensation(12);
-    mRight1.enableVoltageCompensation(12);
-    mRight2.enableVoltageCompensation(12);
-    mLeft1.setIdleMode(IdleMode.kBrake);
-    mLeft2.setIdleMode(IdleMode.kBrake);
-    mRight1.setIdleMode(IdleMode.kBrake);
-    mRight2.setIdleMode(IdleMode.kBrake);
-    mLeft2.follow(mLeft1);
-    mRight2.follow(mRight1);
+    intializeDriveSubystem(mLeft1, mLeft2, mLeft3);
+    intializeDriveSubystem(mRight1, mRight2, mRight3);
+
+    // mLeft2.setClosedLoopRampRate(0.2);
+    // mRight1.setClosedLoopRampRate(0.2);
+    // mRight2.setClosedLoopRampRate(0.2);
     differentialDrive = new DifferentialDrive(mLeft1, mRight1);
 
 
     l_pidController = mLeft1.getPIDController();
     r_pidController = mRight1.getPIDController();
 
     l_encoder = mLeft1.getEncoder();
-    r_encoder = mLeft1.getEncoder();
-
-    kP = 0.00010; 
-    kI = 0;
-    kD = .0000; 
-    kIz = 0; 
-    kFF = 0.000175; 
-    kMaxOutput = 1; 
-    kMinOutput = -1;
+    r_encoder = mRight1.getEncoder();
+    l_encoder.setPosition(0);
+    r_encoder.setPosition(0);
 
     l_pidController.setP(kP);
```

Comment:
Why are we only setting the kP and kI of the left controller? We probably need to set that for both controllers?  I don't think updatePID will set it at startup correctly?

Reply:
we arent, we are also setting right